src/cpp/src/visual_language/embedding_model.hpp

```diff
@@ -61,7 +61,10 @@ class EmbeddingsModel {
         return std::make_shared<EmbeddingsModel>(model, weights, scale_emb, device, properties);
     }
 
-    ov::Tensor infer(const ov::Tensor& input_idx, bool return_remote_tensor=false);
+    // We have getter for the request queue, so we can reserve request outside of infer scope
+    // Tensor produced by infer is stored in the request and used further in the pipeline, so we can't free it right after infer call
+    std::unique_ptr<CircularBufferQueue<EmbeddingsRequest>>& get_request_queue();
```

Comment:
I think users of that class does not actually need unique_ptr - you don't want give them right to eg. reset ptr. Change to returning reference & extract address of queue in places where needed.

CircularBufferQueueElementGuard shouldn't actually take ptr but just reference to queue, since it is not supposed to do anything related to ptr (reset, delete etc). @dkalinowski

Reply:
Sounds good, but since it's not a blocker I would rather postpone making such change and have it merged to master. 
Current commit on this PR has been somewhat checked in terms of accuracy and under heavy load giving good results and I would prefer not rerunning the whole thing if possible.